src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -301,7 +305,7 @@ public ResponseEntity<OnlineCourseConfiguration> updateOnlineCourseConfiguration
     @PostMapping("courses/{courseId}/enroll")
     @EnforceAtLeastStudent
     public ResponseEntity<User> enrollInCourse(@PathVariable Long courseId) {
```

Comment:
do we need the entire domain object here?

Reply:
I would not consider this part of this PR.
I immediately created #7042.